pkg/odo/cli/logs/logs.go

```diff
@@ -0,0 +1,158 @@
+		out: log.GetStdout(),
+	}
+}
+
+var logsExample = ktemplates.Examples(`
+	# Show logs of all containers
+	%[1]s
+`)
+
+func (o *LogsOptions) SetClientset(clientset *clientset.Clientset) {
+	o.clientset = clientset
+}
+
+func (o *LogsOptions) Complete(cmdline cmdline.Cmdline, args []string) error {
+	var err error
+	o.contextDir, err = os.Getwd()
+	if err != nil {
+		return err
+	}
+	isEmptyDir, err := location.DirIsEmpty(o.clientset.FS, o.contextDir)
+	if err != nil {
+		return err
+	}
+	if isEmptyDir {
+		return errors.New("this command cannot run in an empty directory, run the command in a directory containing source code or initialize using 'odo init'")
+	}
+
+	o.Context, err = genericclioptions.New(genericclioptions.NewCreateParameters(cmdline).NeedDevfile(""))
+	if err != nil {
+		return fmt.Errorf("unable to create context: %v", err)
+	}
+
+	o.componentName = o.Context.EnvSpecificInfo.GetDevfileObj().GetMetadataName()
+
+	o.clientset.KubernetesClient.SetNamespace(o.Context.GetProject())
+
+	return nil
+}
+
+func (o *LogsOptions) Validate() error {
+	return nil
+}
+
+func (o *LogsOptions) Run(ctx context.Context) error {
+	containersLogs, err := o.clientset.LogsClient.DevModeLogs(o.componentName, o.Context.GetProject())
+	if err != nil {
+		return err
+	}
+
+	uniqueContainerNames := map[string]struct{}{}
+	for _, entry := range containersLogs {
+		for container, logs := range entry {
+			uniqueName := getUniqueContainerName(container, uniqueContainerNames)
+			uniqueContainerNames[uniqueName] = struct{}{}
+			err = printLogs(uniqueName, logs, o.out)
+			if err != nil {
+				return err
+			}
+		}
+	}
+	return nil
+}
+
+func getUniqueContainerName(name string, uniqueNames map[string]struct{}) string {
+	if _, ok := uniqueNames[name]; ok {
+		// name already present in uniqueNames; find another name
+		// first check if last character in name is a number; if so increment it, else append name with 1
+		last, err := strconv.Atoi(string(name[len(name)-1]))
+		if err == nil {
+			last++
+			name = fmt.Sprintf("%s%d", name[:len(name)-1], last)
+		} else {
+			last = 1
+			name = fmt.Sprintf("%s%d", name, last)
+		}
+		return getUniqueContainerName(name, uniqueNames)
+	}
+	return name
+}
+
+// printLogs prints the logs of the containers with container name prefixed to the log message
+func printLogs(containerName string, rd io.ReadCloser, out io.Writer) error {
+	color.Set(log.ColorPicker())
+	defer color.Unset()
+	scanner := bufio.NewScanner(rd)
+	scanner.Split(bufio.ScanLines)
+
+	for scanner.Scan() {
+		line := scanner.Text()
+		_, err := fmt.Fprintln(out, containerName+": "+line)
+		if err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+func NewCmdLogs(name, fullname string) *cobra.Command {
+	o := NewLogsOptions()
+	logsCmd := &cobra.Command{
+		Use:   name,
+		Short: "Show logs of all containers of the component",
+		Long: `odo logs shows logs of all containers of the component, whether they are running in Dev mode or Deploy mode.
```

Comment:
This description is misleading. Currently, this command only shows logs from containers running in Dev mode.

Reply:
I purposefully kept it this way at this particular place, because I didn't want to have to remember to modify it when support for showing Deploy mode containers' logs was added to `odo logs`. Kinda taking liberty of being in the alpha/beta stage. If you look at the [documentation added in this PR](https://deploy-preview-5760--odo-docusaurus-preview.netlify.app/docs/3.0.0/command-reference/logs/), it doesn't mention Deploy mode anywhere.

If you insist on rewording it to be accurate, I can do so, but I would prefer not to.

Reply:
> I purposefully kept it this way at this particular place, because I didn't want to have to remember to modify it when support for showing Deploy mode containers' logs was added to odo logs.

We could release the next odo alpha/beta before the PR that will complete the implementation is merged, and then even more people would see that misleading help message. 

The odo users should be always on the first place. We should not cut corners to make it easier for ourself and create it harder for users to use odo.

If you are afraid that you will forget to update it, you can keep checklist of the things that you need to update in the original issue. 


>If you look at the [documentation added in this PR](https://deploy-preview-5760--odo-docusaurus-preview.netlify.app/docs/3.0.0/command-reference/logs/), it doesn't mention Deploy mode anywhere.

And that is part of the problem. You correctly omitted that from the docs but kept it in the help message, this will be confusing for users. 

>  Kinda taking liberty of being in the alpha/beta stage. 

We already have a good amount of people experimenting with odo v3, yes we are in alpha/beta stage we can still afford to break stuff, but saying in help message that it works in one way, but in reality not working is not good even for beta stage.


We don't have to merge complete implementation in one PR, but help messages and docs should always match what is actually implemented.

Reply:
Updated.